src/core_api/db/node_database.py

```
@@ -0,0 +1,99 @@
+
+    def _check_indexes(self):
+        try:
+            self.collection.create_index("node_id", unique=True)
+        except PyMongoError as node_db_error:
+            raise RuntimeError(f"Index unresponsive: {node_db_error}")
+        # check indexing for node_id
+
+
+    def create_node(self, node_data: Dict[str, Any]) -> Node:
+        try:
+            node = Node(**node_data)
+            result = self.collection.insert_one(node.model_dump())
+            if not result.acknowledged:
+                raise RuntimeError("Failed to create node")
+            return node
+        except DuplicateKeyError:
+            raise ValueError(f"Node with ID {node_data.get('node_id')} already exists")
+        except PyMongoError as node_db_error:
+            raise RuntimeError(f"Creating operation failed: {node_db_error}")
+        # creating new node
+
+
+    def get_node(self, node_id: str) -> Optional[Node]:
+        try:
+            data = self.collection.find_one({"node_id": node_id})
+            return Node(**data) if data else None
+        except PyMongoError as node_db_error:
+            raise RuntimeError(f"Database operation failed: {node_db_error}")
+        # read node by id
+
+    def get_all_nodes(self) -> List[Node]:
+        try:
+            return [Node(**data) for data in self.collection.find()]
+        except PyMongoError as node_db_error:
+            raise RuntimeError(f"Database operation failed: {node_db_error}")
+        # read all nodes
+
+
+    def update_node(self, node_id: str, update_data: Dict[str, Any]) -> Optional[Node]:
+        try:
+            existing = self.get_node(node_id)
+            if not existing:
+                return None
+
+            updated_data = {**existing.model_dump(), **update_data}
+            result = self.collection.update_one(
+                {"node_id": node_id},
+                {"$set": updated_data}
+            )
+            
+            if result.modified_count == 0:
```

This should raise an exception instead of returning None